src/librustdoc/html/render/print_item.rs

```diff
@@ -643,6 +643,25 @@ fn item_function(cx: &Context<'_>, it: &clean::Item, f: &clean::Function) -> imp
     })
 }
 
+/// Struct used to handle insertion of "negative impl" marker in the generated DOM.
```

Comment:
```suggestion
/// Struct used to handle insertion of "negative impl" marker in the generated DOM.
///
/// This marker appears up to once in an impl list to divide negative impls from positive impls.
```

Currently the only explanation of how the negative impl marker works is buried in `main.js`, this location should be much more accessible.